app/controllers/schools/participants_controller.rb

```diff
@@ -131,4 +137,49 @@ def participant_mentor_form_params
   def email_used?
     User.where.not(id: @profile.user.id).where(email: @profile.user.email).any?
   end
+
+  def active_participant_profiles
+    @school_cohort.active_ecf_participant_profiles
+  end
+
+  def participant_profiles
+    policy_scope(active_participant_profiles)
+  end
+
+  def ineligible_participants
+    policy_scope(active_participant_profiles.ineligible_status.includes(:user).order("users.full_name"))
+  end
+
+  def eligible_participants
+    policy_scope(active_participant_profiles.eligible_status.includes(:user).order("users.full_name"))
+  end
+
+  def contacted_for_info_participants
+    policy_scope(active_participant_profiles.contacted_for_info.includes(:user).order("users.full_name"))
+  end
+
+  def details_being_checked_participants
+    policy_scope(active_participant_profiles.details_being_checked.includes(:user).order("users.full_name"))
+  end
+
+  def fip_participant_categories_feature_flag_active
+    @ineligible = ineligible_participants.without(@eligible)
+    @eligible = eligible_participants
```

Comment:
Doesn't need to happen before this goes in, but before you pick up a new ticket could you extract all this logic to somewhere it can be fully unit tested please? This is pretty complex, and changes elsewhere could very easily break this.

Reply:
I don't mind if it's some kind of service that spits out exactly the groups needed here, or individual components - anything that can have the full range of possible values thrown at is and we can check it does the right thing